src/types/buffer/mod.rs

```diff
@@ -22,43 +22,40 @@ pub trait TypedArray: private::Sealed {
     ///
     /// This may not be used if a mutable borrow is in scope. For the dynamically
     /// checked variant see [`TypedArray::try_borrow`].
-    fn as_slice<'a: 'b, 'b, C>(&'b self, cx: &'b C) -> &'b [Self::Item]
+    fn as_slice<'cx, 'a, C>(&self, cx: &'a C) -> &'a [Self::Item]
     where
-        C: Context<'a>;
+        C: Context<'cx>;
 
     /// Statically checked mutable borrow of binary data.
     ///
     /// This may not be used if any other borrow is in scope. For the dynamically
     /// checked variant see [`TypedArray::try_borrow_mut`].
-    fn as_mut_slice<'a: 'b, 'b, C>(&'b mut self, cx: &'b mut C) -> &'b mut [Self::Item]
+    fn as_mut_slice<'cx, 'a, C>(&mut self, cx: &'a mut C) -> &'a mut [Self::Item]
```

Comment:
The `'a: 'b` bound was unnecessarily complicated. It's not possible to have references that don't meet this requirement.